### ISSUE TITLE
MV inserted rows are taken into account as inserted/affected by insert. This confuses ETL systems.

### ISSUE THREAD

**Issue by den-crane** [ClickHouse/ClickHouse#18237]
```sql
mysql -h 127.0.0.1 -P9004 -u default

create table t(a String) Engine=Memory;

insert into t values(1);
Query OK, 1 row affected (0.001 sec)    <-------------   1 row affected

create materialized view mvt Engine=Memory as select * from t;
insert into t values(1);
Query OK, 2 rows affected (0.001 sec)   <-------------   2 row affected ????
```

```
echo 1|curl -v POST 'http://localhost:8123/?input_format_skip_unknown_fields=1&send_progress_in_http_headers=1&query=INSERT+INTO+t+format+TSV' --data-binary -

< X-ClickHouse-Summary: { .... ,"written_rows":"2"   .....}

```

enterprise ETL systems and my (own) systems check how many rows were inserted to verify consistency, but the current  behavior (number_rows_inserted_into_table+number_rows_inserted_into_MVs) makes this check impossible.

**Comment by edo888** [comment]
Similar problem here. Thanks!

**Comment by den-crane** [comment]
May be we need another metric: parsed_rows ?
A client sends 1000 rows to insert. Clickhouse inserts 32323432 derived rows because of materialized views, input function, etc. and reports "written_rows":"32323432", **"parsed_rows":"1000"**

**Comment by den-crane** [comment]
@arruw this repository is about server. You need https://github.com/ClickHouse/clickhouse-java

**Comment by genzgd** [comment]
To be clear, I believe `clickhouse-java` (and `clickhouse-connect`) both rely on the X-ClickHouse-Summary header for these values,  which is generated by the ClickHouse server.  Agreed it should be documented in both libraries, but the underlying issue is still with ClickHouse server and the summary it generates.

**Comment by TheHett** [comment]
Same problem with JPA when using bathed inserts.

**Comment by harrybarman** [comment]
Yes - this is a major pain for us. As the original comment states the current behavior makes it impossible to check if we've inserted data correctly.
